lib/mix/lib/mix/tasks/deps.unlock.ex

```diff
@@ -27,6 +30,14 @@ defmodule Mix.Tasks.Deps.Unlock do
       opts[:all] ->
         Mix.Dep.Lock.write(%{})
 
+      opts[:check_unused] ->
+        apps = Mix.Dep.load_on_environment([]) |> Enum.map(& &1.app)
+        unused_apps = Mix.Dep.Lock.read() |> Map.drop(apps)
+
+        unless unused_apps == %{} do
+          Mix.shell().error("warning: unused dependencies in mix.lock file")
```

Comment:
We should use `Mix.raise` so the CLI exits with status 1.

Reply:
Done